mr/src/main/java/org/apache/iceberg/mr/mapred/TableResolver.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.mr.mapred;
+
+import java.io.IOException;
+import java.util.Properties;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.hadoop.HadoopTables;
+import org.apache.iceberg.mr.InputFormatConfig;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+final class TableResolver {
+
+  private TableResolver() {
+  }
+
+  static Table resolveTableFromJob(JobConf conf) throws IOException {
+    Properties properties = new Properties();
+    properties.setProperty(InputFormatConfig.CATALOG_NAME,
+        conf.get(InputFormatConfig.CATALOG_NAME, InputFormatConfig.HADOOP_TABLES)); //Default to HadoopTables
+    properties.setProperty(InputFormatConfig.SNAPSHOT_TABLE,
+        conf.get(InputFormatConfig.SNAPSHOT_TABLE, "true"));
+    properties.setProperty(InputFormatConfig.TABLE_LOCATION, extractProperty(conf, InputFormatConfig.TABLE_LOCATION));
+    properties.setProperty(InputFormatConfig.TABLE_NAME, extractProperty(conf, InputFormatConfig.TABLE_NAME));
+    return resolveTableFromConfiguration(conf, properties);
+  }
+
+  static Table resolveTableFromConfiguration(Configuration conf, Properties properties) throws IOException {
+    String catalogName = properties.getProperty(InputFormatConfig.CATALOG_NAME, InputFormatConfig.HADOOP_TABLES);
+    String tableLocation = properties.getProperty(InputFormatConfig.TABLE_LOCATION);
+    String tableName = properties.getProperty(InputFormatConfig.TABLE_NAME);
+    Preconditions.checkNotNull(tableLocation, "Table location is not set.");
+    Preconditions.checkNotNull(tableName, "Table name is not set.");
+    switch (catalogName) {
+      case InputFormatConfig.HADOOP_TABLES:
+        HadoopTables tables = new HadoopTables(conf);
+        if (tableName.endsWith(InputFormatConfig.SNAPSHOT_TABLE_SUFFIX)) {
```

Comment:
According to https://github.com/ExpediaGroup/hiveberg/blob/master/README.md#time-travel-and-system-tables it seems like to access a snapshot metadata table, we need to create a new table which ends in `__snapshots`. Instead of that, can we rather let the user create a table with any name but put the metadata table name as part of the location instead.
```
CREATE TABLE source_db.table_a_snapshot_metadata_table
  STORED BY 'com.expediagroup.hiveberg.IcebergStorageHandler'
  LOCATION 'path_to_original_data_table#snapshots'
    TBLPROPERTIES ('iceberg.catalog'='hadoop.catalog')
```
That way,
1) You don't need to reserve a table suffix
2) You don't need flags like `iceberg.snapshots.table=false`
3) TableResolver code is also simplified as it does not need to understand the syntax for metadata tables in different catalogs. e.g. HadoopTables uses `#snapshots`, HiveCatalog uses `.snapshots`. It simply has to pass the location as-is to the catalog.
4) It will also be able to handle other metadata tables, like partitions, manifest, entries, etc. without any additional handling.

Reply:
I think this is a good idea, especially if it supports all metadata tables without additional work.

Reply:
We've made a note of all of these suggestions so thank you, first of all, and we're working on a new implementation for system tables via Hive. As was [mentioned](https://github.com/apache/iceberg/pull/933#discussion_r439643514) in the other PR, we'll take out all the system table + virtual column stuff from these initial PR's and will open another PR for just system tables after merging this and #933